drivers/gpio/gpio_cc13xx_cc26xx.c

```diff
@@ -22,53 +22,51 @@ struct gpio_cc13xx_cc26xx_data {
 	struct gpio_driver_data common;
 	sys_slist_t callbacks;
 	u32_t pin_callback_enables;
+	u32_t config;
+	u32_t int_config;
 };
 
 static struct gpio_cc13xx_cc26xx_data gpio_cc13xx_cc26xx_data_0;
 
+static int gpio_cc13xx_cc26xx_port_set_bits_raw(struct device *port,
+	u32_t mask);
+static int gpio_cc13xx_cc26xx_port_clear_bits_raw(struct device *port,
+	u32_t mask);
+
 static int gpio_cc13xx_cc26xx_config(struct device *port, int access_op,
 				     u32_t pin, int flags)
 {
-	u32_t config;
+	struct gpio_cc13xx_cc26xx_data *data = port->driver_data;
+	u32_t config = 0;
 
 	if (access_op != GPIO_ACCESS_BY_PIN) {
 		return -ENOTSUP;
 	}
 
+	if (((flags & GPIO_INPUT) != 0) && ((flags & GPIO_OUTPUT) != 0)) {
+		return -ENOTSUP;
+	}
+
+	if (flags == GPIO_DISCONNECTED) {
```

Comment:
We should test individually for `GPIO_INPUT`, `GPIO_OUTPUT` flags, as in:
```
	if (((flags & GPIO_INPUT) == 0) && ((flags & GPIO_OUTPUT) == 0)) {
```

Reply:
Comparing with GPIO_DISCONNECTED seems more readable, and corresponds to what is being handled. I would rather keep it as is unless there is strong objection.

Reply:
This will unfortunately not work since the flag is defined as
```
#define GPIO_DISCONNECTED	0
```
We can only test presence of flags which are defined as `1`.

Reply:
There may be something I am not understanding here. I see this use-case in tests/drivers/gpio/gpio_basic_api/test_gpio_port.c:

rc = gpio_pin_configure(dev, PIN_OUT, GPIO_DISCONNECTED);

The check seems to handle GPIO_DISCONNECTED correctly (the test passes). Do you have any counter example?

Reply:
Actually the intent of `GPIO_DISCONNECTED` was to be able to things like:
```
if ((flags & GPIO_DIR_MASK) == GPIO_DISCONNECTED)
```
which would work just fine.  The only problem with this code is that you can't assume somebody didn't inadvertently include other flags.

Reply:
I see, got it. In my mind `GPIO_DISCONNECTED` was a special flag that you throw on its own. I'll go ahead and change it to the way you have it here.

Reply:
I'll make the same change on the cc32xx as well.

Reply:
FWIW you can see the idiom I was using for mode determination in #19557.  Though don't put `default` first.

Reply:
The `GPIO_DIR_MASK` macro is part of the old flag set, it's currently in the deprecated section and is going to be removed. We shouldn't use it. Currently we have two independent flags `GPIO_INPUT` and `GPIO_OUTPUT` which indicate if the input and respectively output direction should be enabled. The `GPIO_DISCONNECTED` doesn't quite fit the design. Otherwise, we should have used a bit field called e.g. `GPIO_DIR` encoding 4 values: `GPIO_DIR_DISCONNECTED`, `GPIO_DIR_IN`, `GPIO_DIR_OUT`, `GPIO_DIR_INOUT`. But this design doesn't play well with the DTS since in the DTS input and output are configured independently.

Reply:
I was indeed wondering why we didn't have `GPIO_DIR_INOUT`. Thanks for the explanation.

I can replace `GPIO_DIR_MASK` with (GPIO_INPUT | GPIO_OUTPUT). Is it alright if I keep using `GPIO_DISCONNECTED`?

Reply:
> I can replace `GPIO_DIR_MASK` with `(GPIO_INPUT | GPIO_OUTPUT)`. 

Yes, sounds good.

> Is it alright if I keep using `GPIO_DISCONNECTED`?

I would rather avoid using `GPIO_DISCONNECTED` in the driver code, it's really only for the application code to make it easier to read. Again, it would be OK if we used a bit field but instead we have two independent flags.